lms/static/coffee/src/instructor_dashboard/membership.coffee

```diff
@@ -670,6 +670,564 @@ class AuthList
 
+class EmailSelectors
+  constructor: (@$container, @$section, params={}) ->
+    params = _.defaults params,
+      label : $container.data 'label'
+
+    template_html = $("#email-list-widget-template").html()
+    @$container.html Mustache.render template_html, params
+    @cur_column = 0
+    @$table = $("#queryTableBody")
+    @labelArray = ($container.data 'selections').split '<>'
+    @$list_selector = @$container.find 'select.single-email-selector'
+    # populate selectors
+    @$list_selector.empty()
+    @$list_endpoint = $container.data 'list-endpoint'
+    @$rolename = $container.data 'rolename'
+    @$list_selector.append $ '<option/>'
+    if this.$container.attr('data-label') == 'Select Section'
+      @load_list()
+    else if this.$container.attr('data-label') == 'Select Problem'
+      @load_list()
+    else
+      for label in @labelArray
+        @$list_selector.append $ '<option/>',
+          text: label
+
+    @$list_selector.change =>
+      $opt = @$list_selector.children('option:selected')
+      return unless $opt.length > 0
+
+      if this.$container.attr('data-label') == "Select a Type"
+        @chosen_class = $opt.text().trim()
+        if (@chosen_class == "Section")
+          @$section.find('.problem_specific').removeClass("active")
+          @$section.find('.section_specific').addClass("active")
+        else if (@chosen_class == "Problem")
+          @$section.find('.section_specific').removeClass("active")
+          @$section.find('.problem_specific').addClass("active")
+
+  get_list: (cb)->
+    $.ajax
+      dataType: 'json'
+      url: @$list_endpoint
+      data: rolename: 'instructor'
+      success: (data) -> cb? null, data['data']
+      error: std_ajax_err ->
+        cb? gettext("Error fetching problem or section data")
+
+  # load section/problem data
+  load_list: ->
+    @get_list (error, section_list) =>
+      # abort on error
+      return @show_errors error unless error is null
+      _.each section_list, (section) =>
+        @add_row(section, "section")
+        _.each section.sub, (subsection) =>
+          @add_row(subsection , "subsection")
+
+  add_row: (node, useClass) ->
+    @idArr = [node.block_type, node.block_id]
+    @idSt = @idArr.join("/")
+    @toDisplay = node.display_name
+    if node.parents
+      @toDisplay = [node.parents,@toDisplay].join("<>")
+    #indenting subsections with dashes for readability
+    if useClass == "subsection"
+      @toDisplay = "---" + @toDisplay
+    if @toDisplay.length > 50
+      #displaying the last n characters
+      @toDisplay = "..." + @toDisplay.substring(@toDisplay.length-60, @toDisplay.length)
+    @$list_selector.append $ '<option/>',
+            text: @toDisplay
+            class: useClass
+            id : @idSt
+
+  set_cell: (text, colNumber, cellid) ->
+    rows = $("#queryTableBody")[0].rows
+    cell = rows[rows.length - 1].children[colNumber]
+    if cell
+      cell.innerHTML = text
```

Comment:
To clarify, what is the content of this `text` variable?

If it's plaintext, we should use the form `$cell.text(text)`.

If it's actually HTML, we should rename the variable to be more accurate.